### PR TITLE
Add build workflows to the repo

### PR DESCRIPTION
**Pull request by nadheesh**
## Purpose
Resolves https://github.com/wso2-enterprise/choreo/issues/21508

### CODE REVIEW

File: ballerina/Module.md [hunk 1]
```diff
@@ -158,7 +158,7 @@ To create an agent, you need an LLM model and a set of Tool (or ToolKit) definit
 
 
 ```ballerina
-agent:Agent agent = check new (LLMModel model, (ToolKit|Tool)... tools);
+agent:Agent agent = agent:Agent agent = check new (model, ...tools);
```
Comment (NipunaRanasinghe) [inline ballerina/Module.md:161]: is this new code snippet correct?

File: .github/workflows/.github/workflows/build-with-bal-test-native.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+                ACCESS_TOKEN: ${{ secrets.ACCESS_TOKEN }}
+                ORG_NAME: ${{ secrets.ORG_NAME }}
+                USER_REPO_NAME: ${{ secrets.REPO_NAME }}
+                ISSUE_ASSIGNEE: ${{ secrets.ASSIGNEE }}
+                RESOURCE_PATH: ${{ secrets.RESOURCE_PATH }}
+                JAVA_OPTS: -DBALLERINA_DEV_COMPILE_BALLERINA_ORG=true
+                GITHUB_TOPIC: ${{ secrets.TOPIC }}
+                GITHUB_SECRET: ${{ secrets.SECRET }}
+                GITHUB_CALLBACK: ${{ secrets.CALLBACK }}
+                GITHUB_USERNAME: ${{ secrets.USERNAME }}
```
Comment (NipunaRanasinghe) [inline .github/workflows/.github/workflows/build-with-bal-test-native.yml:45]: Some of these env variables are repo level secrets required to run the module test suite. Let's clean this up and keep only the secrets required to run the agent tests (I guess we'll need an openAI token for that?)

File: .github/workflows/.github/workflows/daily-build.yml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+                ACCESS_TOKEN: ${{ secrets.ACCESS_TOKEN }}
+                ORG_NAME: ${{ secrets.ORG_NAME }}
+                USER_REPO_NAME: ${{ secrets.REPO_NAME }}
+                ISSUE_ASSIGNEE: ${{ secrets.ASSIGNEE }}
+                RESOURCE_PATH: ${{ secrets.RESOURCE_PATH }}
+                JAVA_OPTS: -DBALLERINA_DEV_COMPILE_BALLERINA_ORG=true
+                GITHUB_TOPIC: ${{ secrets.TOPIC }}
+                GITHUB_SECRET: ${{ secrets.SECRET }}
+                GITHUB_CALLBACK: ${{ secrets.CALLBACK }}
+                GITHUB_USERNAME: ${{ secrets.USERNAME }}
```
Comment (NipunaRanasinghe) [inline .github/workflows/.github/workflows/daily-build.yml:43]: Some of these env variables are repo level secrets required to run the module test suite. Let's clean this up and keep only the secrets required to run the agent tests (I guess we'll need an openAI token for that?)

Comment (nadheesh) [reply .github/workflows/.github/workflows/daily-build.yml:43]: addressed

File: .github/workflows/.github/workflows/build-with-bal-test-native.yml (hunk 2 above)
Comment (nadheesh) [reply .github/workflows/.github/workflows/build-with-bal-test-native.yml:45]: addressed

File: ballerina/Module.md (hunk 1 above)
Comment (nadheesh) [reply ballerina/Module.md:161]: fixed

File: .github/workflows/.github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  release:
+    types: [published]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Ballerina Build
+        uses: ballerina-platform/ballerina-action/@2201.2.1
```
[the remark this answers is not in the document]
Comment (nadheesh) [reply .github/workflows/.github/workflows/release.yml:13]: fixed

File: .github/workflows/.github/workflows/ci.yml [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+                  ACCESS_TOKEN: ${{ secrets.ACCESS_TOKEN }}
+                  ORG_NAME: ${{ secrets.ORG_NAME }}
+                  USER_REPO_NAME: ${{ secrets.REPO_NAME }}
+                  ISSUE_ASSIGNEE: ${{ secrets.ASSIGNEE }}
+                  RESOURCE_PATH: ${{ secrets.RESOURCE_PATH }}
+                  JAVA_OPTS: -DBALLERINA_DEV_COMPILE_BALLERINA_ORG=true
+                  GITHUB_TOPIC: ${{ secrets.TOPIC }}
+                  GITHUB_SECRET: ${{ secrets.SECRET }}
+                  GITHUB_CALLBACK: ${{ secrets.CALLBACK }}
+                  GITHUB_USERNAME: ${{ secrets.USERNAME }}
```
[the remark this answers is not in the document]
Comment (nadheesh) [reply .github/workflows/.github/workflows/ci.yml:41]: addressed

**Review by NipunaRanasinghe — APPROVED** [review]
LGTM

**Pull request merged**